tests/samples/test_api.py

```diff
@@ -1352,3 +1352,42 @@ async def test_download_artifact(error, tmp_path, spawn_job_client, pg):
     assert (
         get_config_from_app(client.app).data_path / "samples" / "foo" / "fastqc.txt"
     ).read_bytes() == await resp.content.read()
+
+
+@pytest.mark.apitest
+@pytest.mark.parametrize(
+    "update_group, group_settings, all_settings",
+    [
+        (True, False, False),
+        (False, True, False),
+        (False, False, True),
+        (True, True, True),
+    ],
+    ids=["group", "group_settings", "all_settings", "all"],
+)
+async def test_change_rights(
```

Comment:
Please change this to a class-based test with methods for the four cases.

I know I have used parametrization a lot previously, but I have discovered that the tests are difficult to grok and update after writing.